### ISSUE TITLE
Custom breadcrumbs don't show any details in replays

### ISSUE THREAD

**Issue by p-himik** [getsentry/sentry#51217]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Set up Sentry to store all replays and issues
2. Add a breadcrumb with `category`, `level`, `message`, and `data`
3. Log any issue with Sentry
4. Open the issue page and see how it displays the breadcrumbs
5. Open the associated replay and see that breadcrumbs display only the category with `undefined` underneath

### Expected Result

The replay breadcrumbs should display the same information as the one available on the issue page - i.e. both `message` and `data`. Or at the very least just the `message` if also displaying `data` there makes things too crowded (although being able to expand breadcrumbs interactively would be useful here).

### Actual Result

Here's how a breadcrumb on the issue summary page looks:
![image](https://github.com/getsentry/sentry/assets/4410314/3379db25-cf9f-4380-8b5b-b4d2f5fbff26)

And here's how it looks on the replay page:
![image](https://github.com/getsentry/sentry/assets/4410314/d4b179d1-5a3b-4987-957e-ab18db00137a)

Notice that the message also got capitalized for some reason, but at least for me it's not a huge deal.

### Product Area

Replays

**Comment by bruno-garcia** [comment]
Thanks for raising this. I'll check with the team. I suspect it might be by design due to the smaller space we have but it's fair that it might contain critical information we need to display

**Comment by bruno-garcia** [comment]
@p-himik when you raised this issue were you missing some specific `data` to help you debug things? Could you help us by clarifying what use cases or example breadcrumbs require the `data` information in order to be useful?

Making it look 'the same' is challenging because we have a lot less space in there.

**Comment by p-himik** [comment]
In my particular case, most of the time the `message` would be enough. I can't say anything in advance about `data` since it depends entirely on the `message`.
Re-frame events are similar to Redux actions, but their ID and the data are separate, and I log IDs as messages and the data as, well, data. And I _just_ enabled custom breadcrumbs so there haven't been a real need to peer deep into the data just yet, but I can easily imagine heeding to delve deeper and thus having to open the issue page and the replay page side-by-side given the current Sentry implementation.

> Making it look 'the same' is challenging because we have a lot less space in there.

I wouldn't put visual parity as the goal, but I'd say that data parity is definitely important.
Is it an option to have the whole `data` part collapsed by default for every breadcrumb entry when viewed via Replays? In addition to an always visible `message`, of course.

**Comment by bruno-garcia** [comment]
> Is it an option to have the whole data part collapsed by default for every breadcrumb entry when viewed via Replays? In addition to an always visible message, of course.

That sounds doable. I wonder what @Jesse-Box and @ryan953 think

**Comment by ryan953** [comment]
We for sure can make the message better. No surprise that seeing`undefined` in there is really useless. 

Making the data visible is the harder part, I wouldn't want that to be hidden forever like it is now.
I'm thinking about putting that into the Console tab. We could make it visually distinct with an icon and be able to filter for it with "Log Level: breadcrumb" or something like that.

**Comment by Jesse-Box** [comment]
The UX goal of the breadcrumb list is analogous to chapters on YouTube, where you get enough information to investigate further, but without the whole context being crammed into that view. 

I don't have a better solution than what Ryan is offering right now

**Comment by ryan953** [comment]
Hey @p-himik I've got good news! I've finished refactoring our replay breadcrumb data and now we should be consistently showing custom breadcrumbs in the Console tab when you're watching a replay.

The refactoring means that no matter what kinds of custom breadcrumbs you've got, it's going to get displayed with our object explorer.
It looks like this:
![SCR-20230802-nooy](https://github.com/getsentry/sentry/assets/187460/7e68d530-fefc-4623-9b4e-b08fcdde59d6)
